app/javascript/components/DonationBands/DonationBands.js

```diff
@@ -28,69 +28,56 @@ export class DonationBands extends Component {
   props: Props;
 
   state: {
-    customAmount: string,
+    customAmount?: number,
   };
 
   constructor(props: Props) {
     super(props);
 
-    let customAmount = '';
-    if (props.customAmount) {
-      customAmount = props.customAmount.toString();
-    }
-
     this.state = {
-      customAmount: customAmount,
+      customAmount: props.customAmount,
     };
   }
 
   onButtonClicked(amount: number = 0) {
-    this.setState({ customAmount: '' });
+    this.setState({ customAmount: undefined });
     this.props.selectAmount(amount);
     this.props.proceed();
   }
 
   onInputUpdated(value: string) {
-    let amount = null;
-    // \u00a3 is £, \u20ac is €
-    const match = value.match(/^[$\u20ac\u00a3]*(\d{0,10})/);
-
```

Comment:
This hard coded the currency symbols, which doesn't work when the locale changes (e.g. when the € appears at the end: 1,000 €, will not be matched). I've removed that and the new version takes a string (e.g. `CHF 1,000` and strips all non-numeric characters (`CHF 1,000 => 1000`), then parses it. If the user enters only non numeric characters, it will revert to `undefined` and display nothing.